core/src/com/unciv/logic/map/tile/TileImprovementFunctions.kt

```diff
@@ -164,8 +164,11 @@ class TileImprovementFunctions(val tile: Tile) {
                 tile.improvementIsPillaged = false
                 tile.improvement = improvementName
                 improvementFieldHasChanged = true
-
-                removeCreatesOneImprovementMarker()
+                if (improvementName != null && (improvementObject!!.hasUnique(UniqueType.Irremovable) || tile.isMarkedForCreatesOneImprovement(improvementName))) {
```

Comment:
Under what circumstance would we remove an irremovable improvement? This sounds like a case we don't want

Reply:
If we finish a regular tile improvement, this currently resets the MarketForCreatesOneImprovement. So, it can happen you place it, then accidentaly finish a farm or someting on the tile and the District we wanted to place is lost forever. I don't think it should reset it in such a case, but if it is an improvement with Irremovable unique (city center etc.) this should probably take priority.

Reply:
I see :/